akvo/rest/views/typeahead.py

```diff
@@ -59,7 +59,15 @@ def typeahead_user_organisations(request):
 
 @api_view(['GET'])
 def typeahead_project(request):
-    projects = Project.objects.all().exclude(title='')
+    from akvo.rsr.views.project import _project_directory_coll
+    if request.GET.get('published', '0') == '0':
```

Comment:
Come to think of it this is also relevant for EUTF where one of the complaints is that they get projects that aren't "theirs" when searching in MyRSR. At first I thought we might leave it with the fixes you did, but I think we need to be stricter in the project editor too.